scripts/pi-hole/php/teleporter.php

```diff
@@ -574,15 +587,58 @@ function noun($num)
         exit('No file transmitted or parameter error.');
     }
 } elseif (php_sapi_name() === 'cli' && $argc > 1) {
-    // this leaves room for more complicated execution, if such a thing is necessary
-    if (file_exists($argv[1])) {
-        try {
-            load_teleport_archive($argv[1]);  // cli restores everything
-        } catch (PharException $e) {
-            exit($e->getMessage()."\n");
+    try {
+        $arguments = array(
+            'help' => 'print this help message',
+            'whitelist' => 'Load Whitelist (exact) from archive',
+            'regex_whitelist' => 'Load Whitelist (regex/wildcard) from archive',
+            'blacklist' => 'Load Blacklist (exact) from archive',
+            'regexlist' => 'Load Blacklist (regex/wildcard) from archive',
+            'adlist' => 'Load Adlists from archive',
+            'client' => 'Load Clients from archive',
+            'group' => 'Load Groups from archive',
+            'auditlog' => 'Load Audit log from archive',
+            'staticdhcpleases' => 'Load Static DHCP Leases from archive',
+            'localdnsrecords' => 'Load Local DNS Records from archive',
+            'localcnamerecords' => 'Load Local CNAME Records from archive',
+            'flushtables' => 'Clear existing data from a table before loading archive data',
+        );
+        $rest_index = 1;
+        $_POST = getopt('h', array_keys($arguments), $rest_index);
```

Comment:
Please, do not set `$_POST` like this. Use a different variable name.

`$_POST` is a superglobal PHP variable and should only contain information set by a "POST" request. This can make difficult to read or debug the code.

Reply:
Also, you are replacing the original contents of `$_POST`, if any.

Reply:
@iain-henderson 

I re-read all your code.
I get what you are trying to do, but I think creating a "fake" `$_POST` variable manually is not a good idea.

I have some recommendations:

1) do not set `$_POST` values manually.
  The `$_POST` variable is a PHP global variable and should not be changed.
  Creating or changing it can cause confusion in other parts of the page and will make debugging difficult in the future.
  Please, use a different name (I'm using `$args` for this example). 

2) Your original function was using 14 to pass the flags values.
  Now, you are using `$_POST` to decide which flags are enabled, but there is no `$_POST` when using the command line.
  As said on the previous comment, setting `$_POST` values is not desired.
  I suggest you change the function to use a second parameter (an array - I'm using `$flags` for this example):

```php
  function load_teleport_archive($filename, $flags) {
    ...
    $flushtables = isset($flags['flushtables']); 
    $blacklist = isset($flags['blacklist']); 
    $whitelist = isset($flags['whitelist']);
    ...
```

3) Then you can call the function using the second parameter:
```php
   // If called from web interface
   $reloadsettingspage = load_teleport_archive($fullfilename, $_POST);

   ...

   // If called from command line
   load_teleport_archive($argv[$rest_index], $args);
```